### PR TITLE
SDKT-22: Export public API types and create examples for Typescript

### PR DESCRIPTION
**Pull request by tylerlevine**

### CODE REVIEW

File: modules/core/example/ts/create-policy.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const accessToken = '';
+
+// TODO: put the new policy on the wallet with this id
+const id = '';
+
+const coin = 'tltc';
+
+async function main() {
+  bitgo.authenticateWithAccessToken({ accessToken });
+
```
Comment (dannybitgo) [inline modules/core/example/ts/create-policy.ts]: If ```id``` is empty string, I'd print a console message saying "make sure you add your ID to the script"

File: modules/core/example/ts/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## Running the examples
+
+First, run `npm install` in the `modules/core` directory.
+
+```
+$ cd modules/core && npm install 
+```
+
+Then use `npx ts-node` to run the  desired example:
+
```
Comment (dannybitgo) [inline modules/core/example/ts/README.md]: Add ```cd example/ts```

File: modules/core/example/js/create-policy.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-//
-
-const BitGoJS = require('../../src/index.js');
+/**
+ * Create a velocity limit policy on a multi-sig wallet at BitGo.
+ *
+ * This tool will help you see how to use the BitGo API to easily
+ * create new wallet policies
+ *
+ * Copyright 2018, BitGo, Inc.  All Rights Reserved.
```
Comment (dannybitgo) [inline modules/core/example/js/create-policy.js]: 2019

File: modules/core/example/js/create-wallet.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+ * Create a multi-sig wallet at BitGo.
+ * This makes use of the convenience function generateWallet
+ *
+ * This tool will help you see how to use the BitGo API to easily create a wallet.
+ * In this form, it creates 2 keys on the host which runs this example.
+ * It is HIGHLY RECOMMENDED that you GENERATE THE KEYS ON SEPARATE MACHINES for real money wallets!
+ *
+ * To perform more advanced features, such as encrypting keys yourself, please look at createWalletAdvanced.js
+ *
+ * Copyright 2018, BitGo, Inc.  All Rights Reserved.
```
Comment (dannybitgo) [inline modules/core/example/js/create-wallet.js]: 2019

File: modules/core/example/ts/get-maximum-spendable.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ * Get the maximum amount spendable in a single transaction from a wallet.
+ * This value is the sum of the 200 most valuable unspent transactions in your wallet.
+ * This limit exists because an outbound transaction cannot use more than 200 unspent transactions as input,
+ * due to transaction size limits. Note that this value is NOT relevant for account-based coins
+ * i.e. [eth, xrp, xlm], as these coins do not use the unspent transactions data model
+ *
+ * Note that this value may be different than balance, confirmedBalance, and spendableBalance,
+ * which can be obtained using the example in the file: get-wallet-balance.ts
+
+ * Copyright 2018, BitGo, Inc.  All Rights Reserved.
```
Comment (dannybitgo) [inline modules/core/example/ts/get-maximum-spendable.ts]: 2019

File: modules/core/example/ts/create-wallet-addresses.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+const coin = 'tltc';
+const basecoin = bitgo.coin(coin);
+// TODO: set your access token here
+const accessToken = '';
+// TODO: set your passphrase here
+const walletPassphrase = '';
+
```
Comment (dannybitgo) [inline modules/core/example/ts/create-wallet-addresses.ts]: same here, i'd throw an error or add a warning saying "make sure you add your access token / passphrase to the script"

File: modules/core/example/ts/create-wallet.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * To perform more advanced features, such as encrypting keys yourself, please look at createWalletAdvanced.js
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/create-wallet.ts]: warn if empty string

File: modules/core/example/ts/get-maximum-spendable.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+// You can get this from User Settings > Developer Options > Add Access Token
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/get-maximum-spendable.ts]: warn if empty string

File: modules/core/example/ts/get-wallet-balance.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+const coin = 'tltc';
+const basecoin = bitgo.coin(coin);
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/get-wallet-balance.ts]: warn if empty string

File: modules/core/example/ts/get-wallet.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+ * information about a wallet.
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+// You can get this from User Settings > Developer Options > Add Access Token
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/get-wallet.ts]: warn if empty string

File: modules/core/example/ts/get-wallet.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+// TODO: get the wallet with this id
+const id = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/get-wallet.ts]: warn if empty string

File: modules/core/example/ts/list-wallet-shares.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ * This tool will help you see how to use the BitGo API to easily list your
+ * BitGo wallets.
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/list-wallet-shares.ts]: warn if empty string

File: modules/core/example/ts/list-wallet-transfers.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+const coin = 'tltc';
+const basecoin = bitgo.coin(coin);
+// TODO: set your access token here
+const accessToken = '';
+const walletId = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/list-wallet-transfers.ts]: warn if empty string

File: modules/core/example/ts/list-wallets.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+ * This tool will help you see how to use the BitGo API to easily list your
+ * BitGo wallets.
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/list-wallets.ts]: warn if empty string

File: modules/core/example/ts/resend-wallet-share-invite.ts [hunk 15]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Resend the invitation email for sharing a BitGo wallet
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/resend-wallet-share-invite.ts]: warn if empty string

File: modules/core/example/ts/send-wallet-transaction.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo, Coin } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+const coin = 'tltc';
+const basecoin = bitgo.coin(coin) as Coin.Ltc;
+// TODO: set your access token here
+const accessToken = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/send-wallet-transaction.ts]: warn if empty string

File: modules/core/example/ts/share-wallet.ts [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+ * Copyright 2019, BitGo, Inc.  All Rights Reserved.
+ */
+import { BitGo } from '../../src';
+const bitgo = new BitGo({ env: 'test' });
+
+// TODO: set your access token here
+const accessToken = '';
+
+// TODO: set the id of the wallet to share
+const walletId = '';
```
Comment (dannybitgo) [inline modules/core/example/ts/share-wallet.ts]: warn if empty string

File: modules/core/example/ts/create-policy.ts (hunk 1 above)
Comment (tylerlevine) [reply modules/core/example/ts/create-policy.ts]: I'd rather not add this to every example - it's extra boilerplate that doesn't help to understand how to use the API. This is not meant to be production code.

Comment (dannybitgo) [reply modules/core/example/ts/create-policy.ts]: ok fair enough. how about just adding one line to the README.md before ```cd example/ts``` that says something like "update the example script with the required TODO's" ?

Comment (tylerlevine) [reply modules/core/example/ts/create-policy.ts]: Good idea, will update

**Pull request merged**